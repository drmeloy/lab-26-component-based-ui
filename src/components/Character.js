import React, { Component } from 'react';
import Image from './Image';
import Info from './Info';
import PropTypes from 'prop-types';


export default class Character extends Component {
  static propTypes = {
    image:PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    status:PropTypes.string.isRequired,
    species:PropTypes.string.isRequired,
    gender:PropTypes.string.isRequired,
    origin:PropTypes.string.isRequired,
    lastLocation:PropTypes.string.isRequired,
    id: PropTypes.string.isRequired
  }

  render() {
    const { image, name, id, status, species, gender, origin, lastLocation } = this.props;

    return (
      <>
        <Image image={image} name={name} id={id} />
        <Info status={status} species={species} gender={gender} origin={origin} lastLocation={lastLocation} />
      </>
    );
  }
}