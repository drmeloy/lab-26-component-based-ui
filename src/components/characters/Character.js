import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Character.css';

export default class Character extends Component {
  static propTypes = {
    image:PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    status:PropTypes.string.isRequired,
    species:PropTypes.string.isRequired,
    gender:PropTypes.string.isRequired,
    origin:PropTypes.string.isRequired,
    lastLocation:PropTypes.string.isRequired,
    id: PropTypes.string.isRequired
  }

  render() {
    const { image, name, id, status, species, gender, origin, lastLocation } = this.props;

    return (
      <div class={styles.Character}>
        <div style={{ position: 'relative' }}>
          <img src={image} alt={name} />
          <div className={styles.banner}><h2>{name}</h2><br></br><p>{id}</p></div>
        </div>
        <ul>
          <li>
            <span className={styles.left} >Status</span>
            <span className={styles.right} >{status}</span>
          </li><br />
          <li>
            <span className={styles.left} >Species</span>
            <span className={styles.right} >{species}</span>
          </li><br />
          <li>
            <span className={styles.left} >Gender</span>
            <span className={styles.right} >{gender}</span>
          </li><br />
          <li>
            <span className={styles.left} >Origin</span>
            <span className={styles.right} >{origin}</span>
          </li><br />
          <li>
            <span className={styles.left} >Last Location</span>
            <span className={styles.right} >{lastLocation}</span>
          </li><br />
        </ul>
      </div>
    );
  }
}