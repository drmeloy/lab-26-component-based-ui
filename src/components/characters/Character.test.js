import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';
import img1 from '../assets/38.jpeg';

describe('Character component', () => {
  it('renders Character', () => {
    const wrapper = shallow(<Character image={img1} name="Beth Smith" id="id: 38" status="Alive" species="Human" gender="Female" origin="Earth (C-137)" lastLocation="Earth (C-137)"/>);
    expect(wrapper).toMatchSnapshot();
  });
});