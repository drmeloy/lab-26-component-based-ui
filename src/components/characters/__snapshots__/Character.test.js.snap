// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders Character 1`] = `
<div
  class="Character"
>
  <div
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <img
      alt="Beth Smith"
      src="/path/image"
    />
    <div
      className="banner"
    >
      <h2>
        Beth Smith
      </h2>
      <br />
      <p>
        id: 38
      </p>
    </div>
  </div>
  <ul>
    <li>
      <span
        className="left"
      >
        Status
      </span>
      <span
        className="right"
      >
        Alive
      </span>
    </li>
    <br />
    <li>
      <span
        className="left"
      >
        Species
      </span>
      <span
        className="right"
      >
        Human
      </span>
    </li>
    <br />
    <li>
      <span
        className="left"
      >
        Gender
      </span>
      <span
        className="right"
      >
        Female
      </span>
    </li>
    <br />
    <li>
      <span
        className="left"
      >
        Origin
      </span>
      <span
        className="right"
      >
        Earth (C-137)
      </span>
    </li>
    <br />
    <li>
      <span
        className="left"
      >
        Last Location
      </span>
      <span
        className="right"
      >
        Earth (C-137)
      </span>
    </li>
    <br />
  </ul>
</div>
`;
